### DISCUSSION TITLE
Why does limiting a query by 1 still return a list of documents?

### DISCUSSION THREAD

**Discussion by FXschwartz** [firebase/flutterfire#8011] [Feature request]
If I have a collection reference I am querying, and I limit it by 1, it would be really nice if this returned a single document.

`final licenseAgreements = (await reference.whereIsDefault(isEqualTo: true).limit(1).get()).docs;`

**Comment by tpham3783** [comment]
I don't think it's a good idea to have an API that returns a dynamic type.
Imagine that I have to do a type check everytime I use that call.

I would encapsulate that functionality in a mixin , and call it a day.
